src/taco-utils/utilHelper.ts

```diff
@@ -320,6 +322,39 @@ module TacoUtility {
         }
 
         /**
+         * parses a JSON file which could get modified by user or another tool
+         * handles different encoding
+         * @param {string} path to the json file
+         * @return {any} parsed JSON object
+         */
+        public static parseUserJSON(filePath: string): any {
+            if (fs.existsSync(filePath)) {
+                var contents: Buffer = fs.readFileSync(filePath);
+                // try the simplest path first
+                try {
+                    return JSON.parse(contents.toString());
+                } catch (ex) {
+                    UtilHelper.emptyMethod();
+                }
+
+                // may be this is a UTF-8 with BOM
+                try {
+                    return JSON.parse(iconv.decode(contents, "utf8"));
+                } catch (ex) {
+                    UtilHelper.emptyMethod();
+                }
+                // May be this is a UTF-16 file
+                try {
+                    return JSON.parse(iconv.decode(contents, "utf16"));
+                } catch (ex) {
+                    UtilHelper.emptyMethod();
+                }
+            }
+
+            throw errorHelper.get(tacoErrorCodes.TacoErrorCode.ErrorUserJsonMissingOrMalformed, filePath);
```

Comment:
I don't see much added value with that. "missing or malformed" tells clearly either the file doesn't exist or is missing